examples/ecs-fargate/prisma/infra/index.js

```diff
@@ -16,45 +15,32 @@ async function main() {
     app_name: APP_NAME,
     public_ip: true,
     app_port: PORT,
-    image_tag: 'latest'
+    image_tag: 'latest',
   };
   await prisma.ecs_simplified.upsert({
-    where: { app_name: APP_NAME},
+    where: { app_name: APP_NAME },
     create: data,
     update: data,
   });
 
-  const apply = await prisma.$queryRaw`SELECT * from iasql_apply();`
-  console.dir(apply)
-
-  const repoUri = (await prisma.ecs_simplified.findFirst({
-    where: { app_name: APP_NAME },
-    select: { repository_uri: true }
-  })).repository_uri;
-
-  console.log('Docker login...')
-  console.log(execSync(
-    `aws ecr get-login-password --region ${REGION} | docker login --username AWS --password-stdin ${repoUri}`,
-    { stdio: 'pipe', encoding: 'utf8', }, // Output stdout/err through the parent process
-  ));
-
-  console.log('Building image...')
-  console.log(execSync(
-    `docker build -t ${APP_NAME}-repository ${__dirname}/../app`,
-    { stdio: 'pipe', encoding: 'utf8', },
-  ));
-
-  console.log('Tagging image...')
-  console.log(execSync(
-    `docker tag ${APP_NAME}-repository:latest ${repoUri}:latest`,
-    { stdio: 'pipe', encoding: 'utf8', },
-  ));
-
-  console.log('Pushing image...')
-  console.log(execSync(
-    `docker push ${repoUri}:latest`,
-    { stdio: 'pipe', encoding: 'utf8', },
-  ));
+  const apply = await prisma.$queryRaw`SELECT *
+                                       from iasql_apply();`;
+  console.dir(apply);
+
+  console.log('Using ecr_build to build the docker image and push it to ECR...');
+  const repoId = (await prisma.repository.findFirst({
+    where: { repository_name: `${APP_NAME}-repository` },
+    select: { id: true },
+  })).id.toString();
+  const repoUri = `${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}`;
+  const image = await prisma.$queryRaw`SELECT ecr_build(
+              ${repoUri},
+              ${repoId},
+              './examples/ecs-fargate/prisma/app',
+              ${GH_PAT},
+              ${GITHUB_REF}
+  );`;
+  console.log(image);
```

Comment:
is this logger necessary?

Reply:
None of them are necessary, but it fits in with the current flow where the prisma script shows some of the steps of what it's doing to the user, so I'm alright with it.